iree/turbine/kernel/wave/promotion.py

```diff
@@ -77,7 +77,7 @@ def apply_promotion_pattern(
             # We move CustomOp/Read up to the last write_to_shared_mem S.T
             # all reads from shared mem happens only after all read from globals
             # and write to shared mem happen. Which will minimize lds_barrier count.
-            if last_write_to_shared:
+            if last_write_to_shared and last_write_to_shared.graph == custom_node.graph:
```

Comment:
For education purposes, can you explain a little why we need this for extend attention?

Reply:
Sure, since we have multiple nested region ops in the graph, then the last_write_to_shared
is in a different graph as custom.node. So I added this check to make sure they are always in the same graph. Also added a comment.